### PR TITLE
Various nits and fixes before next publish release

### LINKED ISSUE #374: client.listDBs is throwing Error: Method is not accessible, on `onEnd`
**Describe the bug**
client.listDBs is throwing Error: Method is not accessible, on `onEnd` 
> Uncaught (in promise) Error: Method is not accessible
    onEnd index.ts:793

**To Reproduce**
I've a repo [here](https://github.com/Tbaut/textile-playground) where I mostly follow what you do in your test https://github.com/textileio/js-threads/blob/37d8ef1c3ec7a0748465b77d7e29f0cd1494274d/packages/client/src/index.spec.ts

The repo is a CRA for TS, so just clone, `npm install` (or `yarn`), then `yarn start`.
I do the auth with KeyInfo, so you'll need a `.env` file at the root with:
```bash
REACT_APP_HUB_KEY="<your_key>"
REACT_APP_HUB_SECRET="<your_secret>"
```

This is the flow of commands: https://github.com/Tbaut/textile-playground/blob/master/src/components/Threadtest.tsx

Watching the browser console, I get:
image

**Expected behavior**
To list the dbs

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser Firefox
 - Version 78

**Additional context**
Note that I tried to run the test in `js-threads` to see if the problem was on my side, but I run into errors that I don't understand. Wanted to make this issue before falling into the rabbit hole.
```bash
npm run test:browser

> @textile/threads@0.0.0 test:browser /home/thib/github/textileio/js-threads
> lerna run --stream test:browser

lerna notice cli v3.22.1
lerna info versioning independent
lerna info Executing command in 9 packages: "npm run test:browser"
@textile/context: > @textile/context@0.6.0 test:browser /home/thib/github/textileio/js-threads/packages/context
@textile/context: > polendina --cleanup --timeout 60 src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-store: > @textile/threads-store@0.1.16 test:browser /home/thib/github/textileio/js-threads/packages/store
@textile/threads-store: > polendina --cleanup --timeout 60 src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-id: > @textile/threads-id@0.1.10 test:browser /home/thib/github/textileio/js-threads/packages/id
@textile/threads-id: > polendina --cleanup  src/**/*.spec.ts --webpack-config ../../webpack.test.js
@textile/threads-store:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/store/build ...
@textile/threads-id:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/id/build ...
@textile/context:   polendina ☞   Setting up output directory: /home/thib/github/textileio/js-threads/packages/context/build ...
@textile/threads-id: [
@textile/threads-id:   './src/index.ts\n' +
@textile/threads-id:     "Module not found: Error: Can't resolve '@consento/sync-randombytes' in '/home/thib/github/textileio/js-threads/packages/id/src'\n" +
@textile/threads-id:     "resolve '@consento/sync-randombytes' in '/home/thib/github/textileio/js-threads/packages/id/src'\n" +
@textile/threads-id:     '  Parsed request is a module\n' +
...
```

### LINKED ISSUE #621: API docs are out of date
We need to remove links to the gh pages docs and also remove that branch from the repo.

### LINKED ISSUE #636: Schema types not in sync between threads and textile and threaddb
We need to simplify this. For now, the easy fix is to just accept 'any'.

### LINKED ISSUE #637: Switch to js-multiformats over cids where possible
This can be done here, and in js-textile repo pretty much throughout. See https://github.com/ipfs/js-ipfs/issues/3443 for some context. And also https://github.com/multiformats/js-cid/issues/136.

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Description

This includes various fixes that should make it in before our next Threads release.

Fixes #636
Fixes #621 
Fixes #374 (partly, see also https://github.com/textileio/js-textile/pull/786)
Fixes #637 (partly, was mostly already done)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ## Using Threads
 
-To get started using Threads, check out the [docs site](https://docs.textile.io/) and [API documentation](https://textileio.github.io/js-threads).
+To get started using Threads, check out the [docs site](https://docs.textile.io/).
```
Comment (carsonfarmer) [inline README.md:17]: we rely on docs.textile.io now

File: packages/client/package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "@textile/threads-client-grpc": "^1.0.2",
     "@textile/threads-id": "^0.4.0-alpha.3",
     "@types/to-json-schema": "^0.2.0",
+    "fastestsmallesttextencoderdecoder": "^1.0.22",
```
Comment (carsonfarmer) [inline packages/client/package.json:32]: polyfill for text encoder stuff, same as in js-textile

File: packages/client/src/index.ts [hunk 3]
```diff
@@ -108,7 +109,7 @@ export function getFunctionBody(
  */
 export interface CollectionConfig<W = any, R = W> {
   name: string
-  schema?: JSONSchema3or4 | any // Union type to indicate that JSONSchema is preferred but any works
+  schema?: JSONSchema3or4 | any // JSONSchema is preferred but any works
```
Comment (carsonfarmer) [inline packages/client/src/index.ts:112]: just a docs update, we've already relaxed this type on threaddb

File: packages/client/src/index.ts [hunk 4]
```diff
@@ -436,18 +437,19 @@ export class Client {
 
   /**
    * Lists all known DBs.
-   * @remarks this API is blocked on the Hub. Use `listThreads` when importing Client
-   * from `@textile/hub` as an alternative.
    */
   public listDBs(): Promise<
-    Record<string, pb.GetDBInfoReply.AsObject | undefined>
+    Record<string, { id: string; name?: string } | undefined>
```
Comment (carsonfarmer) [inline packages/client/src/index.ts:442]: see https://github.com/textileio/js-textile/pull/786

File: packages/context/src/index.ts [hunk 5]
```diff
@@ -243,8 +243,7 @@ export class Context implements ContextInterface {
 
   withContext(value?: ContextInterface): this {
     if (value === undefined) return this
-    // Spread to copy rather than reference
-    this._context = value.toJSON()
+    this._context = { ...this._context, ...value.toJSON() }
```
Comment (carsonfarmer) [inline packages/context/src/index.ts:246]: this is a fix, but it might be breaking... we'll have to major bump next release anyway

**Review by carsonfarmer — COMMENTED** [review]
A super quick one for @andrewxhill

**Review by asutula — APPROVED** [review]

**Pull request merged**